### DISCUSSION TITLE
Smart fork caching persist smart contracts

### DISCUSSION THREAD

**Discussion by abanchev** [NomicFoundation/hardhat#5976] [Ideas]
If i fork a chain at block 100, and I access 10 smart contracts they will be cached in memory and on the disk, however if I then fork block 101 the cache is invalidated and it will pull them again from the rpc even though contracts are immutable

can we have a smarter caching that invalidates less data

or am i reading it incorrectly, now upon closer inspection some files are not deleted on cache invalidation

**Comment by fvictorio** [comment]
This is a good idea. We weren't doing this before because of the SELFDESTRUCT opcode, but after [EIP-6780](https://eips.ethereum.org/EIPS/eip-6780) this should be a valid optimization.
